cfme/infrastructure/virtual_machines.py

```diff
@@ -1071,113 +1174,118 @@ class VmAllWithTemplatesDetails(CFMENavigateStep):
     VIEW = InfraVmDetailsView
     prerequisite = NavigateToSibling('All')
 
-    def step(self, *args, **kwargs):
-        sel.click(self.obj.find_quadicon(do_not_navigate=True))
+    def step(self):
+        self.prerequisite_view.sidebar.vmstemplates.tree.click_path('All VMs & Templates')
```

Comment:
That's definitely a needed enhancement for VMEntities.

What I'm getting at here is that we can very easily eliminate a big part of this gap by having separate navigation for VM details and Template details, that use their respective accordion menu to narrow down the entities that we're searching for in the first place.

Additionally the details view for a template is different than that for a VM, and the navigation destination needs to return an accurate view.  This can be done with a ConditionalView though that's based on the title, which has 'Template' or 'Virtual Machine' in it.  The navigation destination returns the wrapper view that collapses to either VM view or Template view.